http-core/src/main/scala/org/apache/pekko/http/impl/model/parser/HeaderParser.scala

```diff
@@ -110,11 +110,11 @@ private[http] object HeaderParser {
   object EmptyCookieException extends SingletonException("Cookie header contained no parsable cookie values.")
 
   def lookupParser(headerName: String, settings: Settings = DefaultSettings): Option[String => HeaderParser#Result] =
-    dispatch.lookup(headerName).map { runner => (value: String) =>
-      import pekko.parboiled2.EOI
+    Some { (value: String) =>
```

Comment:
I changed that to keep the old signatures even if the new one might "lie" because it may still fail later with a `RuleNotFound` exception (only if we are adding ModeledHeaders without accompanying it with the correctly named parser).